core/execution/future/liquidation.go

```diff
@@ -0,0 +1,99 @@
+package future
+
+import (
+	"context"
+	"time"
+
+	"code.vegaprotocol.io/vega/core/events"
+	"code.vegaprotocol.io/vega/core/types"
+	"code.vegaprotocol.io/vega/libs/num"
+	"code.vegaprotocol.io/vega/logging"
+)
+
+func (m *Market) checkNetwork(ctx context.Context, now time.Time) error {
+	// debug
+	// this only returns an error if we couldn't get the price range, incidating no orders on book
+	order, _ := m.liquidation.OnTick(ctx, now)
+	if order == nil {
+		return nil
+	}
+	order.OriginalPrice = num.UintZero().Div(order.Price, m.priceFactor)
+	conf, err := m.matching.SubmitOrder(order)
+	if err != nil {
+		m.log.Panic("Failure after submitting order to matching engine",
+			logging.Order(order),
+			logging.Error(err),
+		)
+	}
+	// no order was submitted
+	if conf == nil {
+		return nil
+	}
+	m.broker.Send(events.NewOrderEvent(ctx, order))
+	m.handleConfirmationPassiveOrders(ctx, conf)
+
+	// no trades...
+	if len(conf.Trades) == 0 {
+		m.checkForReferenceMoves(ctx, conf.PassiveOrdersAffected, false)
+		return nil
+	}
+	// transfer fees to the good party -> fees are now taken from the insurance pool
+	fees, _ := m.fee.GetFeeForPositionResolution(conf.Trades)
+	tresps, err := m.collateral.TransferFees(ctx, m.GetID(), m.settlementAsset, fees)
+	if err != nil {
+		// this is the current behaviour, even if this fails, we need to check for reference moves
+		// FIXME(): we may figure a better error handling in here
+		m.checkForReferenceMoves(ctx, conf.PassiveOrdersAffected, false)
+		m.log.Error("unable to transfer fees for positions resolutions",
+			logging.Error(err),
+			logging.String("market-id", m.GetID()))
+		return err
+	}
```

Comment:
Should we actually return here without doing anything else? that looks wrong, seeing that previously we already matched with the orderbook, and produce trades, skipping all that's after if we were just not able to produce transfer the fees seems wrong.

Reply:
The only reason `TransferFees` would return an error is if we are missing accounts, or were otherwise unable to update the balances, if that's the case, we will run in to the same error (most likely) when performing the MTM settlement after handling the trades. I suppose it's better to replace this with a panic though